tests/test_rds.py

```diff
@@ -709,6 +709,48 @@ def test_rds_db_subnetgroup_unused(self):
 
         self.assertEqual(len(resources), 1, "Resources should be unused")
 
+    def test_rds_modify_db(self):
+        session_factory = self.replay_flight_data("test_rds_modify_db")
+        p = self.load_policy(
+            {
+                "name": "rds-modify-db",
+                "resource": "rds",
+                "filters": [
+                    {"DeletionProtection": True},
+                    {"MasterUsername": "testtest"}
+                ],
+                "actions": [
+                    {
+                        "type": "modify-db",
+                        "update": [
+                            {
+                                "property": 'DeletionProtection',
+                                "value": False
+                            }
+                        ],
+                        "immediate": True
+                    }
+                ],
+            },
+            session_factory=session_factory,
+        )
+        resources = p.run()
+        self.assertEqual(len(resources), 1)
+
+        verify_p = self.load_policy(
```

Comment:
don't use policies to verify policies, use a client and do a describe on the resource in question and validate attributes directly, policies have a bunch of other machinery and augments that can obsfucate whats being tested as well as resulting in additional recorded json for the test as well as making the tests slower through extraneous code execution.